tools/fast/environment_devnet.go

```diff
@@ -0,0 +1,182 @@
+package fast
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"sync"
+
+	cid "github.com/ipfs/go-cid"
+	logging "github.com/ipfs/go-log"
+	iptb "github.com/ipfs/iptb/testbed"
+
+	"github.com/filecoin-project/go-filecoin/address"
+)
+
+// EnvironmentDevnet is a FAST lib environment that is meant to be used
+// when working with kittyhawk devnets.
+type EnvironmentDevnet struct {
+	network  string
+	location string
+
+	log logging.EventLogger
+
+	processesMu sync.Mutex
+	processes   []*Filecoin
+}
+
+// NewEnvironmentDevnet builds an environment that uses deployed infrastructure to
+// the kittyhawk devnets.
+func NewEnvironmentDevnet(network, location string) (Environment, error) {
+	env := &EnvironmentDevnet{
+		network:  network,
+		location: location,
+		log:      logging.Logger("environment"),
+	}
+
+	if err := os.MkdirAll(env.location, 0775); err != nil {
+		return nil, err
+	}
+
+	return env, nil
+}
+
+// GenesisCar provides a url where the genesis file can be fetched from
+func (e *EnvironmentDevnet) GenesisCar() string {
+	uri := url.URL{
+		Host:   fmt.Sprintf("genesis.%s.kittyhawk.wtf", e.network),
```

Comment:
Can we factor out the specific hostname to a parameter or struct member? Ditto at line 158.

Reply:
I don't understand the immediate benefit in doing this. Can you elaborate on what use case this would enable?

Reply:
The use case is that someone who is not us could use this code, or we could use it against a network that was not specifically at this domain, e.g. on some local or other infrastructure. I think this is an unnecessary coupling to Protocol Labs' infrastructure.

Reply:
I'd gladly accept PRs when someone setups up their own network and wants to use this code, or if they have something custom they can use the interface and create there own environment. This one is for the kittyhawk devnets.